### DISCUSSION TITLE
How can I include preformatted json in an expand/collapse section?

### DISCUSSION THREAD

**Discussion by erlendthune** [facebook/docusaurus#8756] [Q&A] [answered]
Not sure if this is a bug: When I include the code below in a md file, it results in an error. How can I include json code in an expand/collapse section? I got the idea to use html details/summary from here: https://stackoverflow.com/questions/60827828/docusaurus-v2-collapsible-section-in-markdown

**Steps to reproduce:**
1. Put the code below in an md file:
```
<details><summary>CLICK ME</summary>
<p>
{
    "name": "string"
}
</p>
</details>
```
**Expected result**
<details><summary>CLICK ME</summary>
<p>
{
    "name": "string"
}
</p>
</details>

**Actual result in docusaurus:**
```
Compiled with problems:X

ERROR in ./docs/json.md

Module build failed (from ./node_modules/@docusaurus/mdx-loader/lib/index.js):
SyntaxError: /my-website/json.md: Unexpected token, expected "}" (144:10)

  142 | <p>
  143 | {
> 144 |     "name": "string"
      |           ^
  145 | }
  146 | </p>
  147 | </details>
```

**Comment by pranabdas** [comment]
How about wrapping the json content inside code block?

![Capture](https://user-images.githubusercontent.com/31024886/224287164-72a7624b-8bdb-4759-9da1-3930ae336fef.PNG)

**Reply by erlendthune** [reply]
It does not work unfortunately. You get the same error.

**Reply by pranabdas** [reply]
Did you remove the `<p>` tag, it seems to work for me 🤔

**Reply by erlendthune** [reply]
It does not work on my side. See steps to reproduce below. I just realized markdown is not allowed inside html tags, at least not according to https://daringfireball.net/projects/markdown/syntax#html, and it does not look like even GitHUB supports that either, as you can see if you click on the buttons below.

## Markdown backticks is not supported inside html tags
<details><summary>Click here to see that GitHUB DOES NOT supports json or markdown in general inside details/summary</summary>
```json
{
    "name": "string"
}
```
**bold text**
</details>

## Markdown outside html tags works in GitHUB (and Docusaurus):

```json
{
    "name": "string"
}
```
**bold text**

## Code tag inside details/summary in GitHUB
<details><summary>Click here to see that GitHUB supports json inside a code tag inside details/summary</summary>
<code>
{
    "name": "string"
}
</code>
</details>

**So I guess the problem is really that Docusaurus does not support the code tag inside a details/summary tag.**

# Here is how to reproduce the problem in Docusaurus:

1. Follow the Docusaurus installation instructions, .i.e run 

    npx create-docusaurus@latest my-website classic

2. In the my-website/docs folder, edit one of the md files, f.ex. intro.md and add the code below, f.ex. below Tutorial intro:
![docusaurus1](https://user-images.githubusercontent.com/1288786/224953548-5d88854f-5a13-4485-aea0-44842d87ab86.png)


3. Save the file and run

    npm start

4. Result:

> my-website@0.0.0 start
> docusaurus start

[INFO] Starting the development server...
[SUCCESS] Docusaurus website is running at: http://localhost:3000/

The web browser opens up with this text:

![docusaurus1b](https://user-images.githubusercontent.com/1288786/224954178-c9de9752-a7d6-4cfa-878c-c5bc5939f573.png)

5. Try using the code tag instead:

![docusaurus2](https://user-images.githubusercontent.com/1288786/224953581-f8691984-b985-4f6f-96e2-2b2e6d0f2236.png)

6. The compiler complains about this and does not try to launch the web browser:

![docusaurus3](https://user-images.githubusercontent.com/1288786/224953700-023ba473-e890-42a5-9e12-7caebfb4794f.png)

**Comment by pranabdas** [comment, marked as answer]
Could you please try line breaks before and after the code block?

**Reply by erlendthune** [reply]
You mean like actually adding white space line breaks in the file? I tried that, but it still fails. Does this really work on your side in an md file in Docusaurus? Could it be the version of something that is wrong on my side?

**Reply by pranabdas** [reply]
Yes, it is working for me


<img width="1257" alt="Screenshot 2023-03-14 at 7 34 43 PM" src="https://user-images.githubusercontent.com/31024886/224989310-b7814fef-cf38-4bf5-8a7f-14b8c01e9993.png">

**Reply by erlendthune** [reply]
Fantastic, **that did work**! The reason it did not work on my first attempt was that I tried adding the line breaks before and after code html tags. That didn't work. Going back to the md code syntax with line breaks before and after the code block, like you suggested, is great workaround! Thanks a lot for your help.


![docussauruslf](https://user-images.githubusercontent.com/1288786/225016343-3710024f-206b-435a-b359-67d0c6fc546f.png)

**Reply by Ralphhotrod** [reply]
Thank for helping  me
